assets/src/edit-story/components/library/libraryProvider.js

```diff
@@ -49,23 +49,40 @@ function LibraryProvider({ children }) {
   const initialTab = MEDIA.id;
   const [tab, setTab] = useState(initialTab);
   const [textSets, setTextSets] = useState({});
+  const renderedTabs = useRef({});
   const insertElement = useInsertElement();
   const { insertTextSet, insertTextSetByOffset } = useInsertTextSet();
 
   const { showElementsTab } = useFeatures();
 
   // Order here is important, as it denotes the actual visual order of elements.
-  const tabs = useMemo(
-    () => [
+  const tabs = useMemo(() => {
+    return [
       MEDIA,
       MEDIA3P,
-      ...(tab === TEXT.id ? [TEXT] : [{ icon: TextIcon, id: 'text' }]),
+      TEXT,
       SHAPES,
-      ...(showElementsTab ? [ELEMS] : []),
+      showElementsTab ? ELEMS : {},
       PAGE_LAYOUTS,
-    ],
-    [showElementsTab, tab]
-  );
+    ].reduce((accTabs, renderTab) => {
```

Comment:
It could be a map but it would need a filter for tabs that are not feature-enabled, for example, like the elements tab. Before the array was statically defined and so things like the "Elements" tab would not be in the array like this:
```
SHAPES,
...(showElementsTab ? [ELEMS] : []),
PAGE_LAYOUTS
```

I liked this for its explicit reference to the tab order and the simple logical check for whether or not to include the tab. `.map.filter` is also explicit and maybe more readable, too, but with reduce I'm just iterating through the "tabs" array once to assign the content of the array in one callback. I chose that approach instead of repeating the way the text tab was conditionally rendered before:

```
...(tab === TEXT.id ? [TEXT] : [{ icon: TextIcon, id: 'text' }])
```

So, I could do it with `.map.filter` or define it literally the way it was originally for the text pane if that's preferred. :)

Reply:
```[MEDIA, MEDIA3P, TEXT, SHAPES, showElementsTab && ELEMS, PAGE_LAYOUTS]
        .filter(Boolean)
        .map(...)```
seems alright